### ISSUE TITLE
[TF/Keras][Conv3D+MaxPool3D] Zero-sized spatial dims cause fatal cuDNN abort (should raise Python InvalidArgumentError instead)

### ISSUE THREAD

**Issue by tinywisdom** [tensorflow/tensorflow#101409]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

tensorflow version: 2.20.0

### Custom code

Yes

### OS platform and distribution

Ubuntu 22.04.4 LTS (x86_64)

### Mobile device

Ubuntu 22.04.4 LTS (x86_64)

### Python version

3.10.16

### CUDA/cuDNN version

12.5.1/9.3.0

### Current behavior?

A minimal Keras model stacks `Conv3D` (valid padding) followed by `MaxPool3D` with `pool_size=(1,3,3)` and `strides=(1,2,2)`.
For an input reshaped to `(N=1, D=1, H=10, W=10, C=1)`, the convolution reduces H/W to 2, and the subsequent max pool computes zero spatial sizes for H and W. Instead of raising a Python-side shape error, TensorFlow/cuDNN aborts with:
```
F .../xla/stream_executor/cuda/cuda_dnn.cc:664] Check failed:
cudnnSetTensorNdDescriptor(...) == CUDNN_STATUS_SUCCESS (2000 vs. 0)
batch_descriptor: {count: 1 feature_map_count: 64 spatial: 1 0 0 ...}
Aborted (core dumped)
```
This is surprising and hard to debug; users should get a clear `InvalidArgumentError` (e.g., “pooled output spatial dims become <= 0”).

### Standalone code to reproduce the issue

```shell
import tf_keras as keras
import tensorflow as tf

class LayerModule(keras.layers.Layer):
    def __init__(self, in_features, out_features, activation='relu', **kwargs):
        super().__init__(**kwargs)
        self.dense = keras.layers.Dense(out_features, input_shape=(in_features,))
        self.activation = keras.layers.Activation(activation)
    def call(self, inputs):
        return self.activation(self.dense(inputs))

class BranchingModule(keras.layers.Layer):
    def __init__(self, stem, branch1, branch2, **kwargs):
        super().__init__(**kwargs)
        self.stem, self.branch1, self.branch2 = stem, branch1, branch2
    def call(self, inputs):
        x = self.stem(inputs)
        x = keras.activations.relu(x)
        split_at = x.shape[-1] // 2
        b1_out = self.branch1(x[..., :split_at])
        b2_out = self.branch2(x[..., split_at:])
        pred_weight = tf.reduce_mean(b1_out, axis=0, keepdims=True)  # (1, 10)
        out = tf.matmul(b2_out, pred_weight, transpose_b=True)       # (1, 1)
        return out

class ParamBufferModule(keras.layers.Layer):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.p = self.add_weight(shape=(1, 100), initializer='random_normal', trainable=True)
    def call(self, inputs):
        return tf.matmul(inputs, self.p)  # (1, 1) @ (1, 100) -> (1, 100)

class ConvModule(keras.layers.Layer):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.conv = keras.layers.Conv3D(
            filters=64, kernel_size=(1, 7, 7), strides=(1, 2, 2),
            padding='valid', data_format='channels_last', use_bias=False
        )
        self.relu = keras.layers.ReLU()
        self.maxpool = keras.layers.MaxPool3D(
            pool_size=(1, 3, 3), strides=(1, 2, 2), padding='valid'
        )
    def call(self, inputs):
        x = self.conv(inputs)   # D,H,W: 1,2,2
        x = self.relu(x)
        x = self.maxpool(x)     # D,H,W: 1,0,0  <-- triggers cuDNN abort
        return x

class MyModel(keras.Model):
    def __init__(self):
        super().__init__()
        self.layer1 = LayerModule(100, 100)
        self.sequential = keras.Sequential([LayerModule(100, 100), LayerModule(100, 100)])
        self.pre_branch = keras.layers.Dense(10)
        self.stem = keras.layers.Dense(10)
        self.branch1 = keras.Sequential([keras.layers.Dense(10), keras.layers.ReLU(), keras.layers.Dense(10)])
        self.branch2 = keras.Sequential([keras.layers.Dense(10), keras.layers.ReLU(), keras.layers.Dense(10)])
        self.branching = BranchingModule(self.stem, self.branch1, self.branch2)
        self.param_buffer = ParamBufferModule()
        self.conv_module = ConvModule()

    def call(self, inputs, training=False):
        x = self.layer1(inputs)
        x = self.sequential(x)
        x = self.pre_branch(x)
        x = self.branching(x)              # (1, 1)
        x = self.param_buffer(x)           # (1, 100)
        x = tf.reshape(x, (-1, 1, 10, 10, 1))  # (N, D, H, W, C) = (1,1,10,10,1)
        return self.conv_module(x)

def GetInput():
    return tf.random.uniform(shape=(1, 100))

def trigger():
    model = MyModel()
    x = GetInput()
    # eager path
    _ = model(x)  # will hit the cuDNN abort
    print("This line is not reached if cuDNN aborts")

if __name__ == "__main__":
    trigger()
```

**Comment by Kayyuri** [comment]
Hi @tinywisdom,
Thank you so much for filing this issue! We truly appreciate your support. At the moment, this request is not prioritized and mark it for community contributions. If you would like to contribute on a fix, please feel free to fork the repository and submit a pull request in accordance with the TensorFlow [CONTRIBUTION GUIDELINES](https://www.tensorflow.org/community/contribute/code) and link this issue in your PR to ensure the maintainers can easily review and collaborate.
To avoid duplicate work, please leave a comment here if you decide to take it on!

Thank you again for your interest and we look forward to seeing your contributions!

**Comment by Kayyuri** [comment]
Hi @tinywisdom,
Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

I tried to reproduce the provided code in the Google colab with latest `tf-nightly` version and did not encounter any issue. Kindly refer to this [gist](https://colab.sandbox.google.com/gist/Kayyuri/cf1ac5c90c9e7157e09e410659b7da1d/101409.ipynb) file for your reference and Please let me know if I missed anything.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.
Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
